controllers/cinderapi_controller.go

```diff
@@ -253,20 +253,11 @@ func (r *CinderAPIReconciler) reconcileDelete(ctx context.Context, instance *cin
 	// It's possible to get here before the endpoints have been set in the status, so check for this
 	if instance.Status.APIEndpoints != nil {
 		for _, ksSvc := range keystoneServices {
-			ksSvcSpec := keystonev1.KeystoneServiceSpec{
-				ServiceType:        ksSvc["type"],
-				ServiceName:        ksSvc["name"],
-				ServiceDescription: ksSvc["desc"],
-				Enabled:            true,
-				APIEndpoints:       instance.Status.APIEndpoints[ksSvc["name"]],
-				ServiceUser:        instance.Spec.ServiceUser,
-				Secret:             instance.Spec.Secret,
-				PasswordSelector:   instance.Spec.PasswordSelectors.Service,
-			}
-
-			ksSvcObj := keystone.NewKeystoneService(ksSvcSpec, instance.Namespace, map[string]string{}, 10)
 
+			ks, _ := keystone.GetKeystoneServiceWithName(ctx, helper, ksSvc["name"], instance.Namespace)
+			ksSvcObj := keystone.NewKeystoneService(ks.Spec, instance.Namespace, map[string]string{}, 10)
```

Comment:
I know we're already using this pattern elsewhere (as you noted), but there are two things I was just pondering (and I apologize for not asking these earlier in the similar Glance PR):

1. What happens if the `ks` object comes back as `nil`?
2. Also, is it possible that `ks.Spec` is empty even if `ks` is non-`nil`?

Might be something that's only possible if the user is messing around at the CLI and manually deletes the `KeystoneService` CR.

Reply:
I think if we check for the returned `err` we won't ever get `nil`.  And we could also have a connection issue to etcd and have an error.

Reply:
That sounds like a good approach to me re: check for `err`

Reply:
+1, pushing an update soon with the additional error check we discussed in the previous comment. As per @Akrog comment, I guess it's enough to catch a problematic behavior that should be reported.